dotCMS/src/main/java/com/dotmarketing/db/DotRunnableThread.java

```diff
@@ -0,0 +1,65 @@
+package com.dotmarketing.db;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.util.Config;
+
+public class DotRunnableThread extends Thread {
+
+    final List<DotRunnable> listeners;
+
+    public DotRunnableThread(List<DotRunnable> listeners) {
+
+        this.listeners = listeners;
+    }
+
+    @Override
+    public void run() {
+        try {
+            LocalTransaction.wrap(this::_internalRunner);
+        } catch (Exception dde) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)